File: abc/abc311/b.py
# https://atcoder.jp/contests/abc311/submissions/43890984

n, d = map(int, input().split())
s = [input() for _ in range(n)]

check = [True] * d
for i in range(n):
  for j in range(d):
    if s[i][j] == 'x':
      check[j] = False

ans = 0
cnt = 0
check.append(False)

for b in check:
  if b:
    cnt += 1
  else:
    ans = max(ans, cnt)
    cnt = 0

print(ans)

File: abc/abc311/b.java
// https://atcoder.jp/contests/abc311/submissions/43916591

import java.util.Scanner;

class Main {
  static int max(int x, int y) {
    return (x > y ? x : y);
  }

  public static void main(String[] args) {
    Scanner stdIn = new Scanner(System.in);

    int n = stdIn.nextInt();
    int d = stdIn.nextInt();

    int[] cnt = new int[d + 1];
    for(int i = 0; i <= d; i++) cnt[i] = 0;

    for(int i = 0; i < n; i++) {
      String s = stdIn.next();
      
      for(int j = 0; j < d; j++) {
        if(s.charAt(j) == 'o') {
          cnt[j]++;
        }
      }
    }

    int ans = 0;
    int c = 0;
    for(int i = 0; i <= d; i++) {
      if(cnt[i] == n) {
        c++;
      } else {
        ans = max(ans, c);
        c = 0;
      }
    }

    System.out.println(ans);
  }
}
